Assets/Scripts/Upgrades/Game Logic/HandleCoinsUpgrade.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using CodeControl;
+
+public class HandleCoinsUpgrade : MonoBehaviour
+{    
+    double upgrades = 0;
+    double amount;
+    int count;
```

Comment:
The count should be stored in the upgradeStore, not just here on the handleCoinsUpgrade